### PR TITLE
Patch agent2 errors

### LINKED ISSUE #327: [VIS] panic during cost of living = 5
image

## Repro

https://somas2020.github.io/SOMAS2020/#/newrun

Change `costOfLiving` to 5

Run

image

console logs

image

### PR DESCRIPTION
**Pull request by Eirikalb**
# Summary

Fix #327 
## Additional Information

Initialized former nil map disasterHistory, also fixed an out of range index issue in updateGiftConfidence 
## Test Plan

- Tested min function

### CODE REVIEW

File: internal/clients/team2/forage_test.go [hunk 1]
```diff
@@ -24,3 +24,11 @@ func TestOtherHunters(t *testing.T) {
...
 	}
 }
+
+func TestMinInt(t *testing.T) {
+	bufferLen := MinInt(0, MinInt(5, 3))
+	print("Bufferlen %v", bufferLen)
+	if bufferLen != 0 {
+		t.Error()
+	}
+}
```
Comment (lhl2617) [inline internal/clients/team2/forage_test.go]: strange test

Comment (Eirikalb) [reply internal/clients/team2/forage_test.go]: SanityCheckTest, its a minfunction

File: internal/clients/team2/test.go [hunk 2]
```diff
@@ -0,0 +1 @@
+package team2
```
Comment (lhl2617) [inline internal/clients/team2/test.go]: delete this file?

File: internal/clients/team2/client.go [hunk 3]
```diff
@@ -123,6 +123,7 @@ func NewClient(clientID shared.ClientID) baseclient.Client {
 		foragingReturnsHist:  ForagingReturnsHist{},
 		giftHist:             GiftHist{},
 		islandEmpathies:      IslandEmpathies{},
+		disasterHistory:      DisasterHistory{},
```
Comment (lhl2617) [inline internal/clients/team2/client.go:126]: lgtm

File: internal/clients/team2/forage_test.go (hunk 1 above)
Comment (Eirikalb) [reply internal/clients/team2/forage_test.go]: I can add several testcases if you want though

**Review by lhl2617 — APPROVED** [review]
ok, gogogo! #330 shows more logs when failed on website

**Pull request merged**